org.eclipse.dartboard/src/org/eclipse/dartboard/project/DartFileWizard.java

```diff
@@ -0,0 +1,125 @@
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.dartboard.Messages;
+import org.eclipse.dartboard.util.StatusUtil;
+import org.eclipse.jface.operation.IRunnableWithProgress;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.jface.wizard.Wizard;
+import org.eclipse.osgi.util.NLS;
+import org.eclipse.ui.INewWizard;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPage;
+import org.eclipse.ui.PartInitException;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.ide.IDE;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DartFileWizard extends Wizard implements INewWizard {
+	private static final Logger LOG = LoggerFactory.getLogger(DartFileWizard.class);
+
+	private DartFilePage dartFilePage;
+	private IStructuredSelection selection;
+
+	@Override
+	public void init(IWorkbench workbench, IStructuredSelection selection) {
+		this.selection = selection;
+		setWindowTitle(Messages.NewFile_windowTitle);
+		setNeedsProgressMonitor(true);
+	}
+
+	@Override
+	public void addPages() {
+		dartFilePage = new DartFilePage(DartFilePage.class.getSimpleName(), selection);
+		dartFilePage.setTitle(Messages.NewFile_title);
+		dartFilePage.setDescription(Messages.NewFile_description);
+		dartFilePage.setFileExtension("dart"); //$NON-NLS-1$
+		dartFilePage.setFileName("NewDart"); //$NON-NLS-1$
+		addPage(dartFilePage);
+	}
+
+	@Override
+	public boolean performFinish() {
+		final String containerName = dartFilePage.getContainerFullPath().toOSString();
+		final String fileName = dartFilePage.getFileName();
+		IRunnableWithProgress op = monitor -> {
+			try {
+				doFinish(containerName, fileName, monitor);
+			} catch (CoreException e) {
+				throw new InvocationTargetException(e);
+			} finally {
+				monitor.done();
+			}
+		};
+		try {
+			getContainer().run(true, false, op);
+		} catch (InterruptedException e) {
+			return false;
+		} catch (InvocationTargetException e) {
+			StatusUtil.applyToStatusLine(dartFilePage, StatusUtil.createError(e.getTargetException()));
+			return false;
+		}
+		return true;
+	}
+
+	private void doFinish(String containerName, String fileName, IProgressMonitor monitor) throws CoreException {
+		// create a dart file
+		monitor.beginTask(NLS.bind(Messages.NewFile_creating, fileName), 2);
+		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+		IResource resource = root.findMember(new Path(containerName));
+		if (!resource.exists() || !(resource instanceof IContainer)) {
+			StatusUtil.throwCoreException(NLS.bind(Messages.NewFile_container_doesnot_exist, containerName));
+		}
+		IContainer container = (IContainer) resource;
+		final IFile file = container.getFile(new Path(fileName));
+		try {
+			String contents = "void main() {\r\n" + "    print('Hello World');\r\n" + "}"; //$NON-NLS-1$//$NON-NLS-2$//$NON-NLS-3$

```

Comment:
This doesn't make much sense to do for all new files.

There will only ever be one file per Dart project, that has a main method. 

I would like to only create the file without default contents for now.
In the future we could add the option to select what kind of class should be created and add the corresponding default content.